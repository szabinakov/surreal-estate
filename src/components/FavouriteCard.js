import React from 'react'
import '../styles/FavouriteCard.css'
import PlaceholderPic from '../PlaceholderPic.png'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {faBed, faBath, faPoundSign, faCity, faHome, faEnvelope, faTrashAlt} from '@fortawesome/free-solid-svg-icons'

const FavouriteCard = ({_id, title, type, bedrooms, bathrooms, city, email, price, removeFavourite}) => {


            return (
                
                <div className='EachFavourite' >
                <img src={PlaceholderPic} alt='Placeholder Property'/>
                <div data-testid='favTitleId' className='titleFav'>
                    {title}
                </div>
                <div data-testid='favTypeId' className='typeFav'>
                    <FontAwesomeIcon className='icon' icon={faHome}/>
                    {type}
                </div>
                <div data-testid='favBedroomsId' className='bedroomsFav'>
                    <FontAwesomeIcon className='icon' icon={faBed}/>
                    {bedrooms}
                </div>
                <div data-testid='favBathroomsId' className='bathroomsFav'>
                    <FontAwesomeIcon className='icon' icon={faBath}/>
                    {bathrooms}
                </div>
                <div data-testid='favPriceId' className='priceFav'>
                    <FontAwesomeIcon className='icon' icon={faPoundSign}/>
                    {price}
                </div>
                <div data-testid='favCityId' className='cityFav'>
                    <FontAwesomeIcon className='icon' icon={faCity}/>
                    {city}
                </div>
                <div data-testid='favEmailId' className='emailFav'>
                    <FontAwesomeIcon className='icon' icon={faEnvelope}/>
                    <a className='FavEmailLink' href={email}>Email</a>
                </div>
                <button onClick={() => removeFavourite(_id)} className='RemoveButton'>
                    <FontAwesomeIcon className='icon' icon={faTrashAlt}/>
                    Remove Favourite
                </button>
            </div>
            )
        
}

export default FavouriteCard