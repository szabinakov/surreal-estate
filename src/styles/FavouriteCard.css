.FavouriteContainer{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
}
.EachFavourite {
    border: 1px solid #FAA307;
    border-radius: 1%;
    margin: 15px;
}
img {
    padding: 3px;
}
.icon {
    text-align: start;
    padding-left: 10px;
    padding-right: 10px;
}

.titleFav,.typeFav, .bedroomsFav, .bathroomsFav, .priceFav, .cityFAv, .emailFav {
    padding: 5px 5px 5px 0px;
}
.emailFav {
    padding: 10px;
    background-color: #9d0208;
    text-align: center;
    font-size: 16px;
    font-weight: bolder;
    color: white;
    border: 2px solid;
}

.titleFav {
    text-align: center;
    padding-bottom: 10px;
    font-weight: bold;
}
.FavEmailLink:visited {
color: white;
}

.emailFav:hover {
    border-color: #FAA307
}