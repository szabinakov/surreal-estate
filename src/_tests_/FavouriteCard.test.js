import React from 'react'
import { render } from '@testing-library/react';
import FavouriteCard from '../components/FavouriteCard';


describe('FavouriteCard', () => {
    const mockData = {
        title: '1 bedroom Flat',
        bedrooms: 1,
        bathrooms: 1,
        city: 'Manchester',
        email: 'email@email.com',
        price: 1200
    }

    it('matches the snapshot', () => { 
        const {asFragment} = render (
        <FavouriteCard
            title={mockData.title}
            bedrooms={mockData.bedrooms}
            bathrooms={mockData.bathrooms}
            city={mockData.city}
            email={mockData.email}
            price={mockData.price}/>)

        expect(asFragment()).toMatchSnapshot()
})

    it('renders the correct props', () => {
        const {getByTestId} = render (
            <FavouriteCard
            title={mockData.title}
            bedrooms={mockData.bedrooms}
            bathrooms={mockData.bathrooms}
            city={mockData.city}
            email={mockData.email}
            price={mockData.price}/>)

        expect(getByTestId('favTitleId')).toHaveClass('titleFav')
        expect(getByTestId('favTypeId')).toHaveClass('typeFav')
        expect(getByTestId('favBedroomsId')).toHaveClass('bedroomsFav')
        expect(getByTestId('favBathroomsId')).toHaveClass('bathroomsFav')
        expect(getByTestId('favPriceId')).toHaveClass('priceFav')
        expect(getByTestId('favCityId')).toHaveClass('cityFav')
        expect(getByTestId('favEmailId')).toHaveClass('emailFav')
    })
})