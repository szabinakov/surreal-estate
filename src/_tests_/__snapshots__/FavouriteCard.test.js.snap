// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteCard matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="EachFavourite"
  >
    <img
      alt="Placeholder Property"
      src="PlaceholderPic.png"
    />
    <div
      class="titleFav"
      data-testid="favTitleId"
    >
      1 bedroom Flat
    </div>
    <div
      class="typeFav"
      data-testid="favTypeId"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-home fa-w-18 icon"
        data-icon="home"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="bedroomsFav"
      data-testid="favBedroomsId"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bed fa-w-20 icon"
        data-icon="bed"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
          fill="currentColor"
        />
      </svg>
      1
    </div>
    <div
      class="bathroomsFav"
      data-testid="favBathroomsId"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bath fa-w-16 icon"
        data-icon="bath"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
          fill="currentColor"
        />
      </svg>
      1
    </div>
    <div
      class="priceFav"
      data-testid="favPriceId"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-pound-sign fa-w-10 icon"
        data-icon="pound-sign"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M308 352h-45.495c-6.627 0-12 5.373-12 12v50.848H128V288h84c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-84v-63.556c0-32.266 24.562-57.086 61.792-57.086 23.658 0 45.878 11.505 57.652 18.849 5.151 3.213 11.888 2.051 15.688-2.685l28.493-35.513c4.233-5.276 3.279-13.005-2.119-17.081C273.124 54.56 236.576 32 187.931 32 106.026 32 48 84.742 48 157.961V224H20c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h28v128H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h296c6.627 0 12-5.373 12-12V364c0-6.627-5.373-12-12-12z"
          fill="currentColor"
        />
      </svg>
      1200
    </div>
    <div
      class="cityFav"
      data-testid="favCityId"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-city fa-w-20 icon"
        data-icon="city"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M616 192H480V24c0-13.26-10.74-24-24-24H312c-13.26 0-24 10.74-24 24v72h-64V16c0-8.84-7.16-16-16-16h-16c-8.84 0-16 7.16-16 16v80h-64V16c0-8.84-7.16-16-16-16H80c-8.84 0-16 7.16-16 16v80H24c-13.26 0-24 10.74-24 24v360c0 17.67 14.33 32 32 32h576c17.67 0 32-14.33 32-32V216c0-13.26-10.75-24-24-24zM128 404c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm128 192c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12V76c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 288c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40z"
          fill="currentColor"
        />
      </svg>
      Manchester
    </div>
    <div
      class="emailFav"
      data-testid="favEmailId"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-envelope fa-w-16 icon"
        data-icon="envelope"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
          fill="currentColor"
        />
      </svg>
      <a
        class="FavEmailLink"
        href="email@email.com"
      >
        Email
      </a>
    </div>
    <button
      class="RemoveButton"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-trash-alt fa-w-14 icon"
        data-icon="trash-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
          fill="currentColor"
        />
      </svg>
      Remove Favourite
    </button>
  </div>
</DocumentFragment>
`;
